dataprofiler/tests/labelers/test_base_model.py

```diff
@@ -96,25 +96,99 @@ def test_set_parameters(self):
+
+        # assert bad label inputs
+        with self.assertRaisesRegex(TypeError, '`label` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label=None)
+
+        with self.assertRaisesRegex(TypeError, '`label` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label=1)
+
+        # assert existing label
+        label = 'NEW_LABEL'
+        mock_model._label_mapping = {'NEW_LABEL': 1}
+        with self.assertWarnsRegex(UserWarning,
+                                   'The label, `{}`, already exists in the '
+                                   'label mapping.'.format(label)):
+            base_model.BaseModel.add_label(mock_model, label)
+
+        # assert bad same_as input
+        with self.assertRaisesRegex(TypeError, '`same_as` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label='test', same_as=1)
+
+        label = 'NEW_LABEL_2'
+        same_as = 'DOES_NOT_EXIST'
+        with self.assertRaisesRegex(ValueError,
+                                    '`same_as` value: {}, did not exist in the '
+                                    'label_mapping.'.format(same_as)):
+            base_model.BaseModel.add_label(mock_model, label, same_as)
+
+        # assert successful add
+        label = 'NEW_LABEL_2'
+        base_model.BaseModel.add_label(mock_model, label)
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2}, mock_model._label_mapping)
+
+        # assert successful add w/ same_as
+        label = 'NEW_LABEL_3'
+        base_model.BaseModel.add_label(mock_model, label, same_as='NEW_LABEL')
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2, 'NEW_LABEL_3': 1},
+            mock_model._label_mapping)
+
     def test_set_label_mapping_parameters(self):
 
         # setup mock
         mock_model = mock.Mock(spec=base_model.BaseModel)
+        mock_model._convert_labels_to_label_mapping.side_effect = \
+            base_model.BaseModel._convert_labels_to_label_mapping
 
         # base case
-        with self.assertRaisesRegex(ValueError,
-                                    "`label_mapping` must be a dict which maps "
-                                    "labels to index encodings."):
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
+            base_model.BaseModel.set_label_mapping(mock_model, label_mapping=None)
+
+        # base case
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
             base_model.BaseModel.set_label_mapping(
                 mock_model, label_mapping=None)
```

Comment:
Am I missing something or is this the same test twice?

Reply:
Also what does "base case" mean in this context?

Reply:
updated, good catch